pipenv/utils.py

```diff
@@ -476,21 +545,45 @@ def venv_resolve_deps(
     clear=False,
     allow_global=False,
     pypi_mirror=None,
+    dev=False,
+    pipfile=None,
+    lockfile=None
 ):
     from .vendor.vistir.misc import fs_str
     from .vendor.vistir.compat import Path, to_native_string, JSONDecodeError
     from .vendor.vistir.path import create_tracked_tempdir
-    from .cmdparse import Script
-    from .vendor.pexpect.exceptions import EOF, TIMEOUT
-    from .vendor import delegator
     from . import resolver
-    from ._compat import decode_output
     import json
 
-    if not deps:
-        return []
-
+    vcs_deps = []
+    vcs_lockfile = {}
+    results = []
+    pipfile_section = "dev_packages" if dev else "packages"
+    lockfile_section = "develop" if dev else "default"
+    vcs_section = "vcs_{0}".format(pipfile_section)
+    vcs_deps = getattr(project, vcs_section, [])
+    if not deps and not vcs_deps:
+        return {}
+
+    if not pipfile:
+        pipfile = getattr(project, pipfile_section, None)
+    if not lockfile:
+        lockfile = project._lockfile[lockfile_section]
     req_dir = create_tracked_tempdir(prefix="pipenv", suffix="requirements")
+    if vcs_deps:
+        with create_spinner(text=fs_str("Pinning VCS Packages...")) as sp:
+            vcs_reqs, vcs_lockfile = get_vcs_deps(
+                project,
+                which=which,
+                clear=clear,
+                pre=pre,
+                allow_global=allow_global,
+                dev=dev,
+            )
+            vcs_deps = [req.as_line() for req in vcs_reqs if req.editable]
+            sp.write(environments.PIPENV_SPINNER_OK_TEXT.format(
+                "Successfully pinned VCS Packages!"
```

Comment:
How about removing this ```Successfully pinned VCS Packages!```, which is showed in the final output?

It will be nice to show intermediate status such as pinning vsc deps during operation, but only show the final result which shows success or detailed error info, without showing the intermediate results.

IMHO, the latter one is a little bit redundant.

```
Locking [dev-packages] dependencies...
✔ Success!
Locking [packages] dependencies...
✔ Success!
Updated Pipfile.lock (ab6d38)!
```

```
Locking [dev-packages] dependencies...
✔ Success!
Locking [packages] dependencies...
✔ Successfully pinned VCS Packages!
✔ Success!
Updated Pipfile.lock (ab6d38)!
```

Reply:
agreed, done.